client/src/shared/hooks/useEditAddUserModal.ts

```diff
@@ -0,0 +1,299 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+/* eslint-disable object-shorthand */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+import { useState } from 'react';
+import API from '@/src/apis';
+import { useRouter } from 'next/router';
+
+export interface UserData {
+  id: number;
+  role: string;
+  email: string;
+  username: string;
+  first_name: string;
+  last_name: string;
+  password: string;
+}
+
+const useEditAddModal = (): any => {
+  const [isAddModalOpen, setIsAddModalOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+
+  const [userData, setUserData] = useState<UserData | null>(null);
+
+  const [id, setId] = useState('');
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [role, setRole] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+
+  const [roleError, setRoleError] = useState<string | undefined>('');
+  const [usernameError, setUsernameError] = useState<string | undefined>('');
+  const [emailError, setEmailError] = useState<string | undefined>('');
+  const [firstNameError, setFirstNameError] = useState<string | undefined>('');
+  const [lastNameError, setLastNameError] = useState<string | undefined>('');
+  const [passwordError, setPasswordError] = useState<string | undefined>('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState<
+  string | undefined
+  >('');
+
+  const handleOpenAddModal = (): void => {
+    setIsAddModalOpen(true);
+  };
+
+  const router = useRouter();
+  const params = router.query;
+
+  const handleOpenEditModal = (id: number): void => {
+    const fetchUserData = async (): Promise<void> => {
+      try {
+        const response = await API.get(`user/${params.company_id}/${id}`);
+        const data = response.data;
+        console.log(data);
+        setId(data.id);
+        setUsername(data.username);
+        setRole(data.role);
+        setEmail(data.email);
+        setFirstName(data.first_name);
+        setLastName(data.last_name);
+        setPassword(data.password);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    void fetchUserData();
+
+    setIsEditModalOpen(true);
+  };
+
+  const handleSubmit = async (
+    e: React.FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    e.preventDefault();
+    if (role === '') {
```

Comment:
you can update this logic by creating a reusable validation form, since you used this more than once (add / edit). In this approach, we can also use the validation for other forms.

* ../utils/formValidation.ts
```
function validateForm(formData) {
  const errors = {};
  if (formData.role === '') {
    errors.role = 'Please select a role.';
  }
  if (!formData.username) {
    errors.username = 'Username cannot be empty.';
  }
  if (formData.username.length === 1) {
    errors.username = 'Username should be at least 2 or more characters.';
  }
  if (!formData.email) {
    errors.email = 'Email cannot be empty.';
  }
  if (!formData.firstName) {
    errors.firstName = 'Firstname cannot be empty.';
  }
  if (formData.firstName.length === 1) {
    errors.firstName = 'Firstname should be at least 2 or more characters.';
  }
  if (!formData.lastName) {
    errors.lastName = 'Lastname cannot be empty.';
  }
  if (formData.lastName.length === 1) {
    errors.lastName = 'Lastname should be at least 2 or more characters.';
  }
  if (formData.password.length < 5) {
    errors.password = 'Password must be at least 5 or more characters.';
  }
  if (formData.password !== formData.confirmPassword) {
    errors.confirmPassword = 'Passwords do not match.';
  }
  return errors;
}
```

* ../pages/users/<:company_id>/list
```
function handleSubmit(event) {
  event.preventDefault();
  const formData = {
    role: event.target.elements.role.value,
    username: event.target.elements.username.value,
    email: event.target.elements.email.value,
    firstName: event.target.elements.firstName.value,
    lastName: event.target.elements.lastName.value,
    password: event.target.elements.password.value,
    confirmPassword: event.target.elements.confirmPassword.value,
  };
  const errors = validateForm(formData);
  if (Object.keys(errors).length === 0) {
    // Submit form
  } else {
    // Display errors
  }
}
```